pkg/allocate/allocate.go

```diff
@@ -36,17 +36,20 @@ func AssignIP(ipamConf types.RangeConfiguration, reservelist []types.IPReservati
 	return net.IPNet{IP: newip, Mask: ipnet.Mask}, updatedreservelist, nil
 }
 
-// DeallocateIP assigns an IP using a range and a reserve list.
-func DeallocateIP(reservelist []types.IPReservation, containerID string) ([]types.IPReservation, net.IP, error) {
-
-	updatedreservelist, hadip, err := IterateForDeallocation(reservelist, containerID, getMatchingIPReservationIndex)
-	if err != nil {
-		return nil, nil, err
+// DeallocateIP removes allocation from reserve list.
+func DeallocateIP(reservelist []types.IPReservation, containerID string) ([]types.IPReservation, net.IP) {
```

Comment:
I see your point but I still don't think an error should be returned. However, I will change the logic a bit and return nil for IP when the allocation is not present. It makes more sense in my opinion. When not found, return the reverlist untouched and nil for IP.

Reply:
it's fine the way you currently have. Don't worry. If we had some unit tests at least to define the desired behavior, it could be arguable. Since we don't, let's just use the simplest thing. 

I'm fine w/ what you have. Really.